### PR TITLE
Implement Training and Saving model in train_model.py

### LINKED ISSUE #3: Implement Training and Model Saving in train_model.py
### Description
In the `train_model.py` file, we have a section marked with `# TODO 1` that requires implementation. This task involves loading training data from sub-folders, appending images and labels to the training set, adding person names to the label-to-name dictionary, training the face recognition model, and saving both the trained model and the label-to-name mapping.

### Expected behavior
- Implement the code within the `# TODO 1` section in `train_model.py` to:
  - Load training data.
  - Traverse each sub-folder (assuming each represents a different person).
  - Append each image to the training set.
  - Append the corresponding label to the labels list.
  - Add the person's name to the label-to-name dictionary using the label as the key.
  - Train the face recognition model.
  - Save the trained model to a .xml file in the 'app/trained_models' folder with the naming convention mentioned in the documentation.
  - Save the label-to-name mapping to a CSV file.

### Additional context
- The code structure and comments are already provided in the file.
- You can locate the specific section that needs implementation by searching for `# TODO 1` in the `train_model.py` file.

Completing this task is essential to train the face recognition model and save it for future use in the project. It is a crucial step for the project's functionality.

***

### NOTE
- You may get bonus points for this task depending upon how you train your models.

### PR DESCRIPTION
**Pull request by MauryaRitesh**
Implemented training and saving of model in train_model.py.

## Description

Solved TODO 1.

Fixes #3 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots (if appropriate):
Screenshot 2023-10-12 005403

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

File: train_model/trained_models/LBPH_face_recognition_model.xml
Comment (sanskaromar) [inline train_model/trained_models/LBPH_face_recognition_model.xml:1]: The naming convention of model name isn't followed.

File: train_model/label_to_name_mapping.csv
Comment (sanskaromar) [inline train_model/label_to_name_mapping.csv:1]: No need for this file. Refer to documentation or README for understanding project structure.

File: train_model/train_model.py [hunk 1]
```diff
@@ -31,13 +31,41 @@ def train_face_recognition_model(algorithm='LBPH'):
     # Append the corresponding label to the labels list
     # Add the person's name to the label-to-name dictionary using the label as the key
     # Your code goes here
+    data_dir = 'train_data'
```
Comment (sanskaromar) [inline train_model/train_model.py]: Did you read the code? This path is already declared above.

File: train_model/train_model.py [hunk 2]
```diff
@@ -31,13 +31,41 @@ def train_face_recognition_model(algorithm='LBPH'):
...
+            if filename.endswith(('.jpg', '.jpeg', '.png', '.bmp')): 
+                image_path = os.path.join(person_dir, filename)
+                image = cv2.imread(image_path, cv2.IMREAD_GRAYSCALE) 
+                
+                if image is not None:
+                    faces.append(image)
+                    labels.append(label)
 
     # Train the face recognition model
+    face_recognizer.train(faces, np.array(labels))
```
Comment (sanskaromar) [inline train_model/train_model.py]: I think we can also do `np.array(faces, dtype='object')`

File: train_model/train_model.py [hunk 3]
```diff
@@ -31,13 +31,41 @@ def train_face_recognition_model(algorithm='LBPH'):
...
 
     # Save the trained model to a .xml file in the app/trained_models folder
     # The name of the file should be according to example presented in docs
 
+    model_filename = f'trained_models/{algorithm}_face_recognition_model.xml'
```
Comment (sanskaromar) [inline train_model/train_model.py]: Wrong model naming convention. Also, give absolute path `app/trained_model/{filename}`.

File: train_model/train_model.py [hunk 4]
```diff
@@ -31,13 +31,41 @@ def train_face_recognition_model(algorithm='LBPH'):
...
+    
+    directory = os.path.dirname(model_filename)
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    face_recognizer.save(model_filename)
+
     # Save label-to-name mapping to a CSV file
+    label_to_name_filename = 'label_to_name_mapping.csv'
```
Comment (sanskaromar) [inline train_model/train_model.py]: The file already exists with different name. Read documentation.

**Review by sanskaromar — CHANGES_REQUESTED** [review]
Please make required changes.

File: train_model/train_model.py (hunk 4 above)
Comment (MauryaRitesh) [reply train_model/train_model.py]: @sanskaromar Sorry, I didn't pay much attention. Can you check now?

**Review by sanskaromar — APPROVED** [review]
🚀
